CHANGELOG.md

```diff
@@ -1,6 +1,63 @@
+still experimental, as the upstream spec is still in flux). Further work is
+needed to make enabling this inside ipfs easy and not require
+recompilation.  For more information on the refactor and libp2p v6.0.0, see
+the [release blog post](https://ipfs.io/blog/39-go-libp2p-6-0-0/).  Between
+libp2p v6.0.0 and v6.0.5, there have been some changes worth pointing out
+as well. The swarm dialer has been improved to have shortened timeouts when
+dialing peers in 'local' subnets. This prevents large dial timeouts from
+wasting dial time when the subnet you are dialing is not accessible. The
+TCP handshake timeout has also been dropped to 5 seconds, improving the
+performance of dials to non-existent addresses. In the DHT, we have fixed
+the query code to put records to the K closest peers we can actually
+connect to, which is a strict superset of the peers we previously put
+records to: The peers we could connect to out of the K closest we learned
+about.  Finally, we are changing the way that people write 'ipfs'
+multiaddrs.  Currently, ipfs multiaddrs look something like
+`/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`.
+But caling them 'ipfs' multiaddrs is a bit misleading, as this is actually
+the multiaddr of a libp2p peer that runs ipfs. Other protocols using libp2p
+right now still have to use multiaddrs that say 'ipfs', without actually
+having anything to do with ipfs. Towards that, we are renaming them to
+'p2p' multiaddrs. Moving forward these addresses will be written as:
+`/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`.
+This release adds support for *parsing* both types of addresses into the
+same binary format, and the binary format is remaining exactly the same. A
+future release will have the ipfs daemon switch to *printing* out addresses
+this way, once a large enough portion of the network has upgraded.
+
+On the ipfs side of things, we've added a few small features. The first of
+which is moving to embedding public keys inside IPNS records. This allows
+lookups to be faster, as we only need to fetch the record itself (and not the
+public key separately), and also fixes an issue where DHT peers wouldnt store a
+record for a peer if they didn't have their public key already. Combined with
+some of the DHT and dialing fixes, this should improve the performance of IPNS
+(once a majority of the network updates).
+
+The second feature added is the automatic inclusion of the git plugin in the
+default build. With this, ipfs can ingest git repositories and other data
+directly, and operate over it. For more information on this, see [the
+go-ipld-git repo](https://github.com/ipfs/go-ipld-git).
+
+Various other changes were merged in this release, including great
+documentation, a good number of smaller bugfixes, refactoring and a good bit
+more. For the details, see the changelog below.
+
+
+## 0.4.16-rc2 2018-07-05
```

Comment:
Shouldn't this be a new section?

Reply:
hrm... yeah i'm not sure what reads best here. 

When we do the actual release, would we have an empty section that just says there was not any changes between rc2 and the release?